### DISCUSSION TITLE
Suffix name with `Component` by default

### DISCUSSION THREAD

**Discussion by bilogic** [livewire/livewire#3316] [3. Feature ideas]
Hi,

The basic idea here is to suffix the class name with `Component` by default when using `php artisan make:livewire`. 

Let me illustrate the reasons with a `Cart` example

1. When I run `php make:model -mc Cart`, Laravel creates a model `Cart`, a controller `CartController` and migration to create a `carts` table by default
2. In some cases, a Facade of `Cart::` may be added and the `Cart::` Facade/Model can be called from the Controller
3. I see Controllers and Livewire Components as equivalents, however, Livewire Components are named `Cart` by default today when using `php make:livewire Cart`
4. The problem here is that we cannot call the Facade/Model by their original names from the `Cart` Component. 
5. Since they all have the same name, one must give way, which becomes an unnecessary decision. If we are experienced, great. What about the new comers?
6. While it is fair to say that not every Controller/Component will have a Facade/Model, this approach does not provide a way to deal with a name clash.
7. It is a rabbit hole from here on, but we can keep it simple: why does Laravel name its Controllers with a suffix, and not without?

Thank you.

**Comment by joshhanley** [comment]
@bilogic I'm not sure what you are referring to, as I just ran `php artisan make:controller Test` and it did not name it `TestController` it only named it `Test`.

The Laravel convention is to manually suffix `Controller` yourself when you are generating it.

In Livewire you may do the same using `php artisan make:livewire TestComponent`.

I personally don't think it should be a default, as I use and name them differently.

What I like to do, is for full page Livewire components that are mapped to directly from a route, I like to append `Page`, so like `UsersListPage`. 
Where as components that should only be used inside other Livewire components, then I append `Component` like `UserRowComponent`.

That way I can clearly distinguish between components setup for route binding versus ones that aren't. 🙂

Hope this helps!

**Reply by bilogic** [reply]
@joshhanley I updated the OP to show how Laravel makes `CartController`

1. And I'm using the `Page` suffix too! Although I further move them into a `Pages` subfolder, again, from copying Laravel's namespace convention of *App/Http*/**Controllers** 🙂
2. Perhaps the issue lies with `make:livewire` command being used to make both `Page` and `Component`
3. If I zoom out, both of us arrived at the same set of "changes"
4. What I'm suggesting is that these "changes" be made default so that new comers have an easier time
5. I certainly believe developers can freely name their files, but an opinionated default would have saved us both the trouble

Thanks for sharing!
